src/components/SingleArticle.jsx

```diff
@@ -1,15 +1,74 @@
-export default function SingleArticle({ article }) {
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { getArticleById, getArticleComments, patchArticle } from "../api";
+import Comments from "./Comments";
+
+export default function SingleArticle({ setIsLoading, isLoading }) {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState({});
+  const [comments, setComments] = useState([]);
+  const [feedbackVotes, setFeedbackVotes] = useState("");
+
+  useEffect(() => {
+    setIsLoading(true);
+    getArticleById(article_id)
+      .then((articleById) => {
+        setArticle(articleById);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [article_id]);
+
+  useEffect(() => {
+    getArticleComments(article_id)
+      .then((comments) => {
+        setComments(comments);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, [article_id]);
+
+  function vote(article_id) {
+    patchArticle(article_id);
+
+    setArticle((currArticle) => {
+      return { ...currArticle, votes: currArticle.votes + 1 };
+    });
+    setFeedbackVotes("Vote received");
+  }
```

Comment:
We could do with more happening here.  What if the patch is unsuccessful?    How should we give the user feedback based on an error coming back from the api?  You can test this by setting your network to offline in the inspector. Currently when offline the votes go up as usual.

Reply:
excellent edit!